src/components/LmsConfigurations/index.test.jsx

```diff
@@ -35,32 +52,44 @@ const badBlackboardResponse = {
   request: { status: 400, statusText: 'Bad Request' },
 };
 
-const formData = new FormData();
-
 REQUIRED_MOODLE_CONFIG_FIELDS.forEach((field) => {
-  formData.append(field, 'testdata');
   moodleResponse.data.results[0][field] = 'testdata';
 });
 
 REQUIRED_CANVAS_CONFIG_FIELDS.forEach((field) => {
-  formData.append(field, 'testdata');
   canvasResponse.data.results[0][field] = 'testdata';
 });
 
 REQUIRED_BLACKBOARD_CONFIG_FIELDS.forEach((field) => {
-  formData.append(field, 'testdata');
   blackboardResponse.data.results[0][field] = 'testdata';
 });
 
+REQUIRED_SUCCESS_FACTOR_CONFIG_FIELDS.forEach((field) => {
+  successFactorsResponse.data.results[0][field] = 'testdata';
+});
+
 const waitForAsync = () => new Promise(resolve => setImmediate(resolve));
 
 describe('<LmsConfigurations /> ', () => {
+  it('get list of forms when all configs return 404s', async () => {
+    LmsApiService.fetchMoodleConfig.mockRejectedValue(notFoundMoodleResponse);
+    LmsApiService.fetchBlackboardConfig.mockRejectedValue(notFoundBlackboardResponse);
+    LmsApiService.fetchCanvasConfig.mockRejectedValue(notFoundCanvasResponse);
+    LmsApiService.fetchSuccessFactorsConfig.mockRejectedValue(notFoundSuccessFactorsResponse);
+
+    const wrapper = mount(<LmsConfigurations enterpriseId="testEnterpriseId" />);
+    await waitForAsync();
+    wrapper.update();
+    expect(wrapper.find('ErrorPage').length).toBe(0);
+    expect(wrapper.find('CollapsibleAdvanced').length).toBeGreaterThan(0);
```

Comment:
I feel like this isn't testing much? It could be displaying multiple copies of each form and we wouldn't know. I'd prefer a test that proves that each form we expect to be there is actually there.

Reply:
That's a fair assessment. The goal was just to verify the logic for displaying the partial/collapsible forms was working as expected (ie when we don't have a config present). I felt like using an exact amount might be incredibly tedious and annoying as the test won't be able to be dynamic and every time we add a new form to the page, this count would have to be manually updated (maybe there is a better way to do this, I'm unsure honestly :()

Finding the underlying components could work. (ie in this particular test we should see every form). I think I avoided that initially since at least 1 underlying form should be present in both cases (ie if we have a SuccessFactor config we should see the SuccessFactors component, but if we have no config, we should also see the form)

I didn't even think of a possibility of multiples of the same form to be present.

Reply:
Yeah, I think a test that confirms each form we want to be there is there would be great, that way, if one of them breaks, we'll know.